### DISCUSSION TITLE
You don't always need components™

### DISCUSSION THREAD

**Discussion by Wulfheart** [tailwindlabs/tailwindcss#1999] [General]
While using tailwindcss and tailwindUI I noticed that it takes time to setup some base style to get up running. I used GitHub's [Primer CSS](https://primer.style/css/) and like in many other frameworks there are some components predefined. As far as I know solves tailwind this issue by stating that one should divide an application in different components. However I personally think that it is absolute viable to have a set of minimal styles which can be applied easily for prototyping purposes, e.g. Buttons.
What do you think about it?

**Comment by merelinguist** [comment]
This is kind of against the Tailwind philosophy. If you’re using React/Vue/similar, then you can create a `Button` component which applies the classes. [Here](https://gist.github.com/merelinguist/bd9b9ae348df85a9ea18dafc3008fbb9) is the first few lines of my `Input` component which I use in my app, and which uses the Tailwind UI styles. This way, you can use the same component throughout the site without having to do too much copy-pasting for common elements. (Note I’ve only included the first lines because of the license but I hope you can get the idea).

**Comment by torian257x** [comment]
whatever works for you dude.
Tailwind is nice, but sometimes, you want to write your own classes I am sure, why not?

"NO YOU HAVE TO DO IT ALL WITH TAILWINDCSS COMPLETE FULL" 
well. If someone thinks that, make a good case. 

I for one, wouldnt see necessarily a problem with using tailwindcss and say, semantic ui or something. Maybe need to prefix some things?

**Reply by merelinguist** [reply]
Yeah! Didn’t meant to be so dogmatic. Obviously you can use the framework however you want. I think if I was just writing HTML (and so couldn’t make components), I might be tempted to add some `button` classes.
